src/View/Helper/HtmlHelper.php

```diff
@@ -346,7 +346,8 @@ public function link($title, $url = null, array $options = [])
     {
         $escapeTitle = true;
         if ($url !== null) {
-            $url = $this->Url->build($url);
+            $url = $this->Url->build($url, $options);
+            $options = array_diff_key($options, ['fullBase' => null]);
```

Comment:
Could this be `unset($options['fullBase'])` as well?

Reply:
Yes, it could be - I looked at what was implemented in `::image()` and copied that.

Reply:
After quickly benchmarking it `unset` is about 80% quicker than `array_diff` in this case - didn't see that one coming :smirk:
Will change accordingly!